File: lib/core/network/api_client.dart
import 'package:dio/dio.dart';
import 'dart:developer';
import 'dart:io';
import 'package:flutter/foundation.dart';

class ApiClient {
  // Singleton pattern
  static final ApiClient _instance = ApiClient._internal();
  factory ApiClient() => _instance;

  late final Dio _dio;

  // ✅ Platformaga qarab URL tanlash
  static String get _baseUrl {
    // Web uchun
    if (kIsWeb) {
      return 'http://192.168.1.6:8000';  // ✅ SIZNING IP
    }
    
    // Mobile uchun
    if (Platform.isAndroid) {
      return 'http://192.168.1.6:8000';  // ✅ SIZNING IP
    }
    
    if (Platform.isIOS) {
      return 'http://192.168.1.6:8000';  // ✅ SIZNING IP
    }
    
    // Default (xatolik bo'lsa)
    return 'http://192.168.1.6:8000';
  }

  ApiClient._internal() {
    _dio = Dio(
      BaseOptions(
        baseUrl: _baseUrl,
        connectTimeout: const Duration(seconds: 10),
        receiveTimeout: const Duration(seconds: 10),
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
        },
      ),
    );

    // Log interceptor
    _dio.interceptors.add(InterceptorsWrapper(
      onRequest: (options, handler) {
        log("🚀 REQUEST[${options.method}] => PATH: ${options.path}");
        log("📍 BASE_URL: ${options.baseUrl}");
        return handler.next(options);
      },
      onResponse: (response, handler) {
        log("✅ RESPONSE[${response.statusCode}] => DATA: ${response.data}");
        return handler.next(response);
      },
      onError: (DioException e, handler) {
        log("❌ ERROR[${e.response?.statusCode}] => MESSAGE: ${e.message}");
        log("📍 URL: ${e.requestOptions.uri}");
        return handler.next(e);
      },
    ));
  }

  /// Telefon raqamni tekshirish (Bot yuborgan token orqali)
  Future<Map<String, dynamic>?> verifyPhone({
    required String token,
    required int telegramId,
    required String phone,
    String? firstName,
    String? lastName,
    String? username,
  }) async {
    const String path = "/api/v1/auth/verify";

    final Map<String, dynamic> payload = {
      "token": token,
      "telegram_id": telegramId,
      "phone": phone,
      "first_name": firstName,
      "last_name": lastName,
      "username": username,
    };

    try {
      final response = await _dio.post(path, data: payload);
      
      if (response.statusCode == 200) {
        return response.data as Map<String, dynamic>;
      }
      return null;
    } on DioException catch (e) {
      log("Dio xatolik: ${e.response?.data ?? e.message}");
      return null;
    } catch (e) {
      log("Kutilmagan xatolik: $e");
      return null;
    }
  }
  
  /// API URL ni o'zgartirish uchun (development vaqtida)
  void updateBaseUrl(String newUrl) {
    _dio.options.baseUrl = newUrl;
    log("🔄 Base URL o'zgartirildi: $newUrl");
  }
}

File: telegram-bot/bot/services/api_client.py
"""
Backend API bilan aloqa
"""
import aiohttp
from typing import Optional, Dict, Any
from bot.config import settings
import logging

logger = logging.getLogger(__name__)


class BackendAPIClient:
    """Backend API client"""
    
    def __init__(self):
        self.base_url = settings.BACKEND_URL
        self.timeout = aiohttp.ClientTimeout(total=30)
    
    async def verify_phone(
        self,
        token: str,
        telegram_id: int,
        phone: str,
        first_name: Optional[str] = None,
        last_name: Optional[str] = None,
        username: Optional[str] = None
    ) -> Optional[Dict[str, Any]]:
        """
        Telefon raqamni backend'ga yuborish
        """
        url = f"{self.base_url}/api/v1/auth/verify"
        
        payload = {
            "token": token,
            "telegram_id": telegram_id,
            "phone": phone,
            "first_name": first_name,
            "last_name": last_name,
            "username": username
        }
        
        try:
            async with aiohttp.ClientSession(timeout=self.timeout) as session:
                async with session.post(url, json=payload) as response:
                    if response.status == 200:
                        data = await response.json()
                        logger.info(f"✅ Phone verified for user {telegram_id}")
                        return data
                    else:
                        error = await response.json()
                        logger.error(f"❌ API Error: {error}")
                        return None
        except aiohttp.ClientError as e:
            logger.error(f"❌ Network error: {e}")
            return None
        except Exception as e:
            logger.error(f"❌ Unexpected error: {e}")
            return None


# Singleton instance
api_client = BackendAPIClient()